app/services/imo_api_service.rb

```diff
@@ -42,27 +65,55 @@ def execute_post(url, data)
       .basic_auth(user: IMO_USERNAME, pass: IMO_PASSWORD)
       .post(url, data)
   rescue HTTP::Error => e
-    raise Error.new("Error while calling the IMO API", path: url, exception_message: e)
+    raise Error.new("Error while calling the Imo API", path: url, exception_message: e)
   end
 
-  def process_response(response, url)
+  def process_response(response, url, action)
     case response.status
-    when 200 then :invited
+    when 200
+      body = JSON.parse(response.body)
+      body_status = body.dig("response", "status")
+      return :invited if body_status == "success" && action == "invitation"
+      return :subscribed if body_status == "success" && action == "notification"
```

Comment:
Imo just added invitation callbacks to their API doc, so this will probably change. Without that callback, the only way to know if they've subscribed is to send a message and see if it succeeds.